### PR TITLE
Subscription filter

### PR DESCRIPTION
**Pull request by ashu555**

### CODE REVIEW

File: app/Http/Controllers/Order/ExtendedOrderController.php [hunk 1]
```diff
@@ -58,12 +61,25 @@ private function getBaseQueryForOrders()
...
-                'subscriptions.update_ends_at as subscription_ends_at', 'subscriptions.id as subscription_id', 'subscriptions.version as product_version', 'subscriptions.updated_at',
+                'subscriptions.update_ends_at as subscription_ends_at', 'subscriptions.id as subscription_id', 'subscriptions.version as product_version', 'subscriptions.created_at', 'subscriptions.updated_at',
                 'products.name as product_name', \DB::raw("concat(first_name, ' ', last_name) as client_name"), 'client as client_id',
                 'users.currency'
             );
     }
 
+    private function installedNotInstalled($installedNotInstalled, $join)
+    {
+        if ($installedNotInstalled) {
```
Comment (avinash403) [inline app/Http/Controllers/Order/ExtendedOrderController.php:72]: simplify this logic

Comment (avinash403) [reply app/Http/Controllers/Order/ExtendedOrderController.php:72]: Add test cases for the same

File: app/Http/Controllers/Order/ExtendedOrderController.php [hunk 2]
```diff
@@ -265,6 +281,50 @@ private function expiryTill($expiry, $expiryTill, $join)
...
+     * Searches for Subcription From Date.
+     *
+     * @param string $expiry The Subcription From Date
+     * @param object $join
+     *
+     * @return Query
+     */
+    private function subFrom($till, $from, $join)
+    {
+        if ($from) {
```
Comment (avinash403) [inline app/Http/Controllers/Order/ExtendedOrderController.php:294]: what happens if from is null? what will it return?
Please try to keep same return type wherever possible

Comment (avinash403) [reply app/Http/Controllers/Order/ExtendedOrderController.php:294]: Add test cases for this

File: app/Http/Controllers/Order/ExtendedOrderController.php [hunk 3]
```diff
@@ -265,6 +281,50 @@ private function expiryTill($expiry, $expiryTill, $join)
...
+
+    /**
+     * Searches for Order Till Date.
+     *
+     * @param string $expiry The Order Till Date
+     * @param object $join
+     *
+     * @return Query
+     */
+    private function subTill($from, $till, $join)
```
Comment (avinash403) [inline app/Http/Controllers/Order/ExtendedOrderController.php:315]: read the above comments for do the same here

File: resources/views/themes/default1/common/dashboard.blade.php [hunk 4]
```diff
@@ -89,7 +89,10 @@
               <i class="fa fa-arrow-circle-right"></i></a>
              </div>
         </div>
-
+         @php
+        $startDate = new Carbon\Carbon('-30 days');
+        $endDate = Carbon\Carbon::now()->subDays(1);
```
Comment (avinash403) [inline resources/views/themes/default1/common/dashboard.blade.php:94]: you can use `subDay` for 1 day

File: resources/views/themes/default1/order/index.blade.php [hunk 5]
```diff
@@ -296,5 +325,11 @@ function checking(e){
     $('#datepicker4').datepicker({
       autoclose: true
     })
+    $('#datepicker5').datepicker({
```
Comment (avinash403) [inline resources/views/themes/default1/order/index.blade.php:328]: you can use class for such kind of scenarios and can give same class to all of them

File: app/Http/Controllers/Order/ExtendedOrderController.php (hunk 1 above)
Comment (ashu555) [reply app/Http/Controllers/Order/ExtendedOrderController.php:72]: Fixed

File: app/Http/Controllers/Order/ExtendedOrderController.php (hunk 2 above)
Comment (ashu555) [reply app/Http/Controllers/Order/ExtendedOrderController.php:294]: When null, it doesn't filters subscription from field.

File: app/Http/Controllers/Order/ExtendedOrderController.php (hunk 3 above)
Comment (ashu555) [reply app/Http/Controllers/Order/ExtendedOrderController.php:315]: Fixed

File: resources/views/themes/default1/common/dashboard.blade.php (hunk 4 above)
Comment (ashu555) [reply resources/views/themes/default1/common/dashboard.blade.php:94]: Fixed

File: resources/views/themes/default1/order/index.blade.php (hunk 5 above)
Comment (ashu555) [reply resources/views/themes/default1/order/index.blade.php]: Fixed

**Pull request merged**